### DISCUSSION TITLE
Application Customizer not executing when registering through Add-PnPCustomAction

### DISCUSSION THREAD

**Discussion by webdevelperify** [pnp/powershell#2276] [General]
Hi Team,

Continuation of this [https://github.com/pnp/PnP-PowerShell/issues/1773](issue).

I am using **PnP Powershell (v1.9.0)** to install the Application Customizer for header and footer logic in SharePoint Online Site. I have build the sppkg file and using PnP Powershell installed this in a single Site Collection. On Installation there is no error but this Application Customizer is not executing in any of the page.

**WebPart Manifest**
```
{
  "$schema": "https://developer.microsoft.com/json-schemas/spfx/client-side-extension-manifest.schema.json",

  "id": "a350253b-30a9-49e7-a221-7c6076e3d302",
  "alias": "HelloWorldApplicationCustomizer",
  "componentType": "Extension",
  "extensionType": "ApplicationCustomizer",

  // The "*" signifies that the version should be taken from the package.json
  "version": "*",
  "manifestVersion": 2,

  // If true, the component can only be installed on sites where Custom Script is allowed.
  // Components that allow authors to embed arbitrary script code should set this to true.
  // https://support.office.com/en-us/article/Turn-scripting-capabilities-on-or-off-1f2c515f-5d7e-448a-9fd7-835da935584f
  "requiresCustomScript": false
}
```

**Command Used:**
```
Add-PnPApp -Path "<path>\sharepoint\solution\sp-fx-extension.sppkg" -Scope Site -Publish
Add-PnPCustomAction -Name "HelloWorldApplicationCustomizer" -Title "HelloWorldApplicationCustomizer" -Location "ClientSideExtension.ApplicationCustomizer" -ClientSideComponentId "a350253b-30a9-49e7-a221-7c6076e3d302" -ClientSideComponentProperties "{""testMessage"":""Test New""}" -Sequence 0
```

After this I see in PowerShell that Custom Action is registered in the Web scope successfully but still it is not executed:
![image](https://user-images.githubusercontent.com/30578236/178626098-bcc80f44-9bd9-4236-8361-44242850b174.png)

I do not want to deploy this sppkg file at the tenant level. I tried to just publish this app at the Tenant app catalog (not installed to all sites) and then used the PowerShell above at one site collection still the result is same and application customizer not executed.

Can anyone please help or guide me. Thank you in advance

**Comment by CallumCrowley** [comment]
@webdevelperify Please use the Bug Report template when reporting bugs. You can do this by clicking New Issue > Get Started (next to Bug Report).

**Comment by gautamdsheth** [comment]
Hi, am converting this to a discussion since it is a question and not an issue with the module itself.

Would suggest that you try this cmdlet:

https://pnp.github.io/powershell/cmdlets/Add-PnPApplicationCustomizer.html
